### PR TITLE
Replace IbisDoc with JavaDoc

### PR DESCRIPTION
**Pull request by ricardovh**

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/http/HttpSenderBase.java [hunk 1]
```diff
@@ -815,45 +814,66 @@ public String getPhysicalDestinationName() {
...
+	 */
 	public void setMethodType(HttpMethod method) {
 		this.httpMethod = method;
 	}
 
 	/**
 	 * This is a superset of mimetype + charset + optional payload metadata.
 	 */
-	@IbisDoc({"Content-Type (superset of mimetype + charset) of the request, for <code>POST</code>, <code>PUT</code> and <code>PATCH</code> methods", "text/html, when postType=<code>RAW</code>"})
+	/**
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/http/HttpSenderBase.java:841]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/pipes/CleanupOldFilesPipe.java [hunk 2]
```diff
@@ -177,63 +176,78 @@ public boolean accept(File file) {
...
 	}
 	public String getFilePattern() {
 		return filePattern;
 	}
 
-	@IbisDoc({"", " "})
+	/**
+	 * 
+	 * @ff.default  
+	 */
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/pipes/CleanupOldFilesPipe.java:190]: Empty?

File: core/src/main/java/nl/nn/adapterframework/pipes/CounterSwitchPipe.java [hunk 3]
```diff
@@ -81,7 +80,10 @@ public int getDivisor() {
 		return divisor;
 	}
 
-	@IbisDoc({" ", "2"})
+	/**
+	 *
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/pipes/CounterSwitchPipe.java:84]: No description

File: core/src/main/java/nl/nn/adapterframework/pipes/SkipPipe.java [hunk 4]
```diff
@@ -80,12 +79,18 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 	/**
 	 * Sets the number of bytes to skip
 	 */
-	@IbisDoc({"Number of bytes (for binary input) or characters (for character input) to skip. An empty byte array or string is returned when skip is larger then the length of the input", "0"})
+	/**
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/pipes/SkipPipe.java:82]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/receivers/FileRecordListener.java [hunk 5]
```diff
@@ -352,15 +351,18 @@ public String getDirectoryProcessedFiles() {
 	/**
 	 * set the time to delay when no records are to be processed and this class has to look for the arrival of a new file
 	 */
-	@IbisDoc({"The time <i>in milliseconds</i> to delay when no records are to be processed, and this class has to look for the arrival of a new file", "1000"})
+	/**
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/receivers/FileRecordListener.java:354]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/receivers/FileRecordListener.java [hunk 6]
```diff
@@ -329,7 +328,7 @@ public String getInputDirectory() {
 	/**
 	 * set the {@link WildCardFilter wildcard} to look for files in the specified directory, e.g. "*.inp"
 	 */
-	@IbisDoc({"the wildcard to look for files in the specified directory, e.g. \"*.inp\"", ""})
+	/** the wildcard to look for files in the specified directory, e.g. \"*.inp\" */
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/receivers/FileRecordListener.java:331]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/receivers/FileRecordListener.java [hunk 7]
```diff
@@ -341,7 +340,7 @@ public String getWildcard() {
 	 * Sets the directory to store processed files in
 	 * @param directoryProcessedFiles The directoryProcessedFiles to set
 	 */
-	@IbisDoc({"the directory to store processed files in", ""})
+	/** the directory to store processed files in */
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/receivers/FileRecordListener.java:343]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/receivers/FileRecordListener.java [hunk 8]
```diff
@@ -318,7 +317,7 @@ public String getName() {
 	 * set the directory name to look in for files.
 	 * @see #setWildcard(String)
 	 */
-	@IbisDoc({"set the directory name to look in for files.", ""})
+	/** set the directory name to look in for files. */
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/receivers/FileRecordListener.java:320]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/senders/FixedResultSender.java [hunk 9]
```diff
@@ -159,12 +161,12 @@ public void setFileName(String fileName) {
 	 *
 	 * @param filename the name of the file to return the contents from
 	 */
-	@IbisDoc({"name of the file containing the resultmessage", ""})
+	/** name of the file containing the resultmessage */
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/senders/FixedResultSender.java:164]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/senders/IbisLocalSender.java [hunk 10]
```diff
@@ -296,18 +295,18 @@ public SenderResult sendMessage(Message message, PipeLineSession session) throws
 	/**
 	 * Sets a serviceName under which the JavaListener or WebServiceListener is registered.
 	 */
-	@IbisDoc({"Name of the {@link WebServiceListener} that should be called", ""})
+	/** Name of the {@link WebServiceListener} that should be called */
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/senders/IbisLocalSender.java:298]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/senders/IbisLocalSender.java [hunk 11]
```diff
@@ -320,30 +319,45 @@ public void setReturnedSessionKeys(String string) {
...
+	 */
 	public void setSynchronous(boolean b) {
 		synchronous = b;
 	}
 
 	/**
 	 * When <code>true</code>, the call is made in a separate thread, possibly using separate transaction.
 	 */
-	@IbisDoc({"If <code>true</code>, the call is made in a separate thread, possibly using separate transaction", "false"})
+	/**
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/senders/IbisLocalSender.java:333]: Double JavaDoc for 1 method, how to handle?

File: core/src/main/java/nl/nn/adapterframework/batch/IRecordHandlerManager.java [hunk 12]
```diff
@@ -41,7 +40,7 @@ public interface IRecordHandlerManager extends INamedObject {
 	/**
 	 * @param flow New flow to be added to the managed flow elements
 	 */
-	@IbisDoc({"Flow to be added to the managed flow elements"})
+	/** Flow to be added to the managed flow elements */
```
Comment (ricardovh) [inline core/src/main/java/nl/nn/adapterframework/batch/IRecordHandlerManager.java:43]: Double JavaDoc for 1 method, how to handle?

**Comment by ricardovh** [comment]
Current issues:
- Pipeline JavaDoc for default success exit does not fill values
- `{@value }` is not parsed by FrankDoc
- `@IbisDocRef` is not replaced yet

File: core/src/main/java/nl/nn/adapterframework/batch/IRecordHandlerManager.java [hunk 13]
```diff
@@ -57,7 +56,7 @@ public interface IRecordHandlerManager extends INamedObject {
 	/**
 	 * @param initialFactory indicates if this manager is the initial manager
 	 */
-	@IbisDoc("indicates if this manager is the initial manager")
+	/** indicates if this manager is the initial manager */
```
[the remark this answers is not in the document]
Comment (gvanbrakel) [reply core/src/main/java/nl/nn/adapterframework/batch/IRecordHandlerManager.java:59]: remove the other javadoc

File: core/src/main/java/nl/nn/adapterframework/pipes/CleanupOldFilesPipe.java (hunk 2 above)
Comment (gvanbrakel) [reply core/src/main/java/nl/nn/adapterframework/pipes/CleanupOldFilesPipe.java:190]: fix or remove

File: core/src/main/java/nl/nn/adapterframework/batch/RecordTransformer.java [hunk 14]
```diff
@@ -281,7 +280,7 @@ public void registerOutputFields(OutputfieldsPart part) throws ConfigurationExce
 	 * @param outputfieldsDef
 	 * @throws ConfigurationException
 	 */
-	@IbisDoc({"semicolon separated list of output record field specifications (see table below)", ""})
+	/** semicolon separated list of output record field specifications (see table below) */
```
[the remark this answers is not in the document]
Comment (gvanbrakel) [reply core/src/main/java/nl/nn/adapterframework/batch/RecordTransformer.java:283]: in general: extract any information that is useful for Frank Developers, and add that to the 'IbisDoc-javadoc', and then remove the old javadoc.

File: core/src/main/java/nl/nn/adapterframework/pipes/CounterSwitchPipe.java (hunk 3 above)
Comment (gvanbrakel) [reply core/src/main/java/nl/nn/adapterframework/pipes/CounterSwitchPipe.java:84]: replace with @Default annotation

File: core/src/main/java/nl/nn/adapterframework/receivers/ExchangeMailListener.java [hunk 15]
```diff
@@ -65,76 +64,76 @@ public void setTempFolder(String tempFolder) {
...
 	}
 
-	@IbisDocRef({ EXCHANGE_FILE_SYSTEM})
+	/** @ff.ref nl.nn.adapterframework.filesystem.ExchangeFileSystem */
 	public void setAuthAlias(String authAlias) {
 		getFileSystem().setAuthAlias(authAlias);
 	}
 
-	@IbisDocRef({ EXCHANGE_FILE_SYSTEM})
+	/** @ff.ref nl.nn.adapterframework.filesystem.ExchangeFileSystem */
```
[the remark this answers is not in the document]
Comment (ricardovh) [reply core/src/main/java/nl/nn/adapterframework/receivers/ExchangeMailListener.java:111]: Will not fix, already existing issues or issues where no JavaDoc is required (@ff.ref)

File: core/src/main/java/nl/nn/adapterframework/senders/ExchangeFolderSender.java [hunk 16]
```diff
@@ -34,32 +33,32 @@ public ExchangeFolderSender() {
...
 	}
 
-	@IbisDocRef({ EXCHANGE_FILE_SYSTEM})
+	/** @ff.ref nl.nn.adapterframework.filesystem.ExchangeFileSystem */
 	public void setClientSecret(String clientSecret) {
 		getFileSystem().setClientSecret(clientSecret);
 	}
 
-	@IbisDocRef({ EXCHANGE_FILE_SYSTEM})
+	/** @ff.ref nl.nn.adapterframework.filesystem.ExchangeFileSystem */
```
[the remark this answers is not in the document]
Comment (ricardovh) [reply core/src/main/java/nl/nn/adapterframework/senders/ExchangeFolderSender.java:56]: Will not fix, already existing issues or issues where no JavaDoc is required (@ff.ref)

File: core/src/main/java/nl/nn/adapterframework/senders/FtpFileSystemSender.java [hunk 17]
```diff
@@ -35,132 +34,132 @@ public FtpFileSystemSender() {
...
 	}
 
-	@IbisDocRef({FTPFILESYSTEM})
+	/** @ff.ref nl.nn.adapterframework.filesystem.FtpFileSystem */
 	public void setPort(int port) {
 		getFileSystem().setPort(port);
 	}
 
-	@IbisDocRef({FTPFILESYSTEM})
+	/** @ff.ref nl.nn.adapterframework.filesystem.FtpFileSystem */
```
[the remark this answers is not in the document]
Comment (ricardovh) [reply core/src/main/java/nl/nn/adapterframework/senders/FtpFileSystemSender.java:52]: Will not fix, already existing issues or issues where no JavaDoc is required (@ff.ref)

**Comment by ricardovh** [comment]
Current issues:
- `{@value }` is not parsed by FrankDoc
- For `@ff.ref` the entire package+classname is now in the JavaDoc, while previously a private static final value was used. These are still there (see Codacy), but FrankDoc only handles full package+classname

**Review by nielsm5 — APPROVED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Pull request merged**